### PR TITLE
Add Cloud region name to ZoneApi

### PR DESCRIPTION
**Pull request by freva**

### CODE REVIEW

File: config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneApi.java [hunk 1]
```diff
@@ -17,4 +17,6 @@
     default RegionName getRegionName() { return getId().region(); }
 
     CloudName getCloudName();
+
+    String getCloudNativeRegionName();
```
Comment (hakonhall) [inline config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneApi.java]: Document what this returns for the supported clouds, AWS for now. It's the AWS availability zone you want with this right?

Comment (freva) [reply config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneApi.java]: No, the AWS region name, e.g. `us-east-1`

File: config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneApi.java [hunk 2]
```diff
@@ -18,5 +18,6 @@
 
     CloudName getCloudName();
 
+    /** Returns the region name within the cloud, e.g. 'us-east-1' */
```
Comment (hakonhall) [inline config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneApi.java]: This comment does not help - more context is needed, e.g. "'us-east-1' for AWS"

Comment (freva) [reply config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneApi.java]: It applies for all current clouds...

Comment (freva) [reply config-provisioning/src/main/java/com/yahoo/config/provision/zone/ZoneApi.java]: Fixed.

**Review by hakonhall — APPROVED** [review]

**Pull request merged**